src/conversation/conversation.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Conversation } from './conversation.entity';
+import { RollbarLogger } from 'nestjs-rollbar';
+
+@Injectable()
+export class ConversationService {
+  private logger: Logger = new Logger();
+  constructor(
+    @InjectRepository(Conversation)
+    private conversationRepository: Repository<Conversation>,
+    private readonly rollbarLogger: RollbarLogger,
+  ) {}
+
+  async create(): Promise<string> {
+    const conversation = new Conversation();
+    conversation.stage = 'setup';
+
+    return await this.conversationRepository
+      .save(conversation)
+      .then(conversation => {
+        return conversation.id;
+      })
+      .catch(error => {
+        this.rollbarLogger.error(error, 'Save Conversation');
+        return null;
+      });
+  }
+
+  async get(conversationId: string): Promise<{ language; stage }>;
+  async get(conversationId: string, column: string): Promise<string>;
+
+  async get(
+    conversationId: string,
+    column?: string,
+  ): Promise<string | { language; stage }> {
+    const conversation = await this.conversationRepository
+      .findOne(conversationId)
+      .catch(error => this.rollbarLogger.error(error, 'Get Conversation'));
+
+    if (!conversation) {
+      this.rollbarLogger.error(
+        {
+          message: 'Cannot find a conversation with id: ' + conversationId,
+          name: 'No ConversationId',
+        },
+        'Get Conversation',
+      );
+    }
+
+    return column ? conversation[column] : conversation;
```

Comment:
If `conversation` was not found, this line will throw an error, or return `null`. We may want to trigger a `404 Not Found` in the controller.

On a more general note, is there an error handling pattern/mechanic in NestJS to automatically handle not found entities from the db? (E.g. in Rails there's the ability for the entity find to throw a specific exception that gets caught higher up the stack and gets handled [and logged/reported] and a `404` response is rendered).

Reply:
NestJs automatically handles errors as they float up the chain to the response so a 404 will automatically happen in this case, but it should be caught by the rollbar error above also.